### PR TITLE
Add GitHub app bearer token generation

### PR DESCRIPTION
**Pull request by casperboone**
This PR adds app bearer token generation based on private keys.

### CODE REVIEW

File: modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Environment properties of the GitHub test reporter.
+ */
+object Properties {
+    val appId: String
+    val appPrivateKeyLocation: String
+
+    init {
+        if (File("githubtestreporter.properties").exists()) {
```
Comment (JSAbrahams) [inline modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt]: Is this something of the `java-jwt` library?

Comment (gandreadis) [reply modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt]: Why should it be? This is simple a file-exists check, right?

Comment (FWDekker) [reply modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt]: But where does the file come from?

File: settings.gradle [hunk 2]
```diff
@@ -14,6 +14,8 @@ rootProject.name = "schaapi"
 
 includeModule "application"
 
+includeModule "github-test-reporter"
```
Comment (FWDekker) [inline settings.gradle]: Are we going to put all these parts in the global mainspace? Doesn't it technically belong to the pipeline?

Comment (gandreadis) [reply settings.gradle]: It does, sounds to me like [step 2.2](https://github.com/cafejojo/schaapi#22-notify-developers) of the pipeline

File: modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt (hunk 1 above)
Comment (casperboone) [reply modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt]: The file is optional, a user can create it

File: settings.gradle (hunk 2 above)
Comment (gandreadis) [reply settings.gradle]: We'd ideally split the pipeline into two (mining and execution), but that's out of scope for this PR. I think the best place for this right now would be the `pipeline` module

File: modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt (hunk 1 above)
Comment (FWDekker) [reply modules/github-test-reporter/src/main/kotlin/org/cafejojo/schaapi/githubtestreporter/Properties.kt]: Could you document that somewhere?

File: settings.gradle (hunk 2 above)
Comment (gandreadis) [reply settings.gradle]: Or, more appropriately, in the `validation-pipeline` folder?

**Comment by casperboone** [comment]
@JSAbrahams See https://github.com/cafejojo/schaapi/pull/160#discussion_r192366995

**Pull request merged**